thcrap_proton

```diff
@@ -71,6 +73,28 @@ if [ ! -f "$THCRAP_FOLDER/config/$THCRAP_CONFIG" ]; then
     fi
 fi
 
+if [ "$USE_THPRAC" == 1 ]; then
+    THPRAC_VERSION=$(LD_LIBRARY_PATH=/usr/lib64:/usr/lib curl -si https://github.com/touhouworldcup/thprac/releases/latest | grep location: | sed 's@.*/@@' | tr -d '[:space:]')
+    if [ ! -f "$THPRAC_FOLDER/thprac.$THPRAC_VERSION.exe" ]; then
+        if ${ZENITY} --question --text="thprac doesn't appear to be installed on \"$THPRAC_FOLDER\". Would you like this script to install it there?"; then
+            mkdir -p "$THPRAC_FOLDER"
+            if [ ! -w "$THPRAC_FOLDER" ]; then
+                ${ZENITY} --error --text="The directory \"$THPRAC_FOLDER\" could not be created or is not writable. Check your permissions and try again."
+                exit
+            fi
+            LD_LIBRARY_PATH=/usr/lib64:/usr/lib curl -L -f "https://github.com/touhouworldcup/thprac/releases/download/$THPRAC_VERSION/thprac.$THPRAC_VERSION.exe" -o "$THPRAC_FOLDER/thprac.$THPRAC_VERSION.exe" | ${ZENITY} --progress --pulsate --auto-close --text="Downloading thprac.\"$THPRAC_VERSION\".exe"
+            if [ $? != 0 ] || [ ! -f "$THPRAC_FOLDER/thprac.$THPRAC_VERSION.exe" ]; then
+                ${ZENITY} --error --text="Download failed. Aborting"
+                exit
+            fi
+            # Create a symlink to latest version
+            ln -srf "$THPRAC_FOLDER/thprac.$THPRAC_VERSION.exe" "$THPRAC_FOLDER/thprac.exe"
+        else
+            USE_THPRAC=0
+        fi
+    fi
+fi
```

Comment:
This will fail when there's no Internet connection. Since it's just an executable and doesn't carry around user config files, we could just install/update it silently. Let's also put a timeout in the first curl to avoid stalling the game launch for too long.

```suggestion
if [ "$USE_THPRAC" == 1 ]; then
    THPRAC_VERSION=$(LD_LIBRARY_PATH=/usr/lib64:/usr/lib curl --connect-timeout 5 -si https://github.com/touhouworldcup/thprac/releases/latest | grep location: | sed 's@.*/@@' | tr -d '[:space:]')
    if [[ ! -z "$THPRAC_VERSION" && ! -f "$THPRAC_FOLDER/thprac.$THPRAC_VERSION.exe" ]]; then
        mkdir -p "$THPRAC_FOLDER"
        if [ ! -w "$THPRAC_FOLDER" ]; then
            ${ZENITY} --error --text="The directory \"$THPRAC_FOLDER\" could not be created or is not writable. Check your permissions and try again."
            exit
        fi
        LD_LIBRARY_PATH=/usr/lib64:/usr/lib curl -L -f "https://github.com/touhouworldcup/thprac/releases/download/$THPRAC_VERSION/thprac.$THPRAC_VERSION.exe" -o "$THPRAC_FOLDER/thprac.$THPRAC_VERSION.exe" | ${ZENITY} --progress --pulsate --auto-close --text="Downloading thprac.$THPRAC_VERSION.exe"
        if [ $? != 0 ] || [ ! -f "$THPRAC_FOLDER/thprac.$THPRAC_VERSION.exe" ]; then
            # Only abort if there's no version currently installed
            if [ ! -f "$THPRAC_FOLDER/thprac.exe" ]; then
                ${ZENITY} --error --text="Download failed. Aborting"
                exit
            fi
        else
            # Create a symlink to latest version
            ln -srf "$THPRAC_FOLDER/thprac.$THPRAC_VERSION.exe" "$THPRAC_FOLDER/thprac.exe"
        fi
    fi
fi
```